v2/pkg/engine/resolve/errors.go

```diff
@@ -0,0 +1,109 @@
+package resolve
+
+import (
+	"fmt"
+	"strings"
+)
+
+type GraphQLError struct {
```

Comment:
There is an [ExternalError](https://github.com/wundergraph/graphql-go-tools/blob/04d41b42261ef2e29c6857a6fe544323011444af/v2/pkg/operationreport/externalerror.go#L31) type in operationreport package, which we probably could reuse to not create the same type again

Reply:
I tried but `ExternalError` is not the representation we need to serialize the error. `Path` is from type `ast.Path` but I need `[]string`

Reply:
hmm, basically we could adjust [RequestError](https://github.com/wundergraph/graphql-go-tools/blob/c281b9c1142317ba50053e8e87287377298f3283/v2/pkg/graphqlerrors/errors.go#L112)
Currently for the path it is using custom struct, but it is not mandatory

If you want I could adjust it for you purpose

Reply:
Please go ahead. Thank you.